framework/src/base/Attributes.C

```diff
@@ -277,6 +277,24 @@ AttribThread::isEqual(const Attribute & other) const
   return isMatch(other);
 }
 
+void
+AttribExecutionOrderGroup::initFrom(const MooseObject * obj)
+{
+  const auto * uo = dynamic_cast<const UserObject *>(obj);
+  _val = uo ? uo->getParam<int>("execution_order_group") : 0;
```

Comment:
need an assert for the cast. People could be using your work elsewhere on non-UOs

Reply:
the `uo` pointer is explicitly checked right below